src/components/YearSelector/index.tsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+import { ArrowBackIosNew, ArrowForwardIos } from '@mui/icons-material';
+import {
+  IconButton,
+  Button,
+  Typography,
+  Popover,
+  ListItemButton,
+} from '@mui/material';
+import { Dispatch, SetStateAction, useState } from 'react';
+
+interface YearSelectorProps {
+  selectedYear: number;
+  onChange: Dispatch<SetStateAction<number>>;
+}
```

Comment:
i totally forgot to mention this, but i think we should use a url query param for this. if we don't we have to either provide a context for components down the tree to use or drill props to where we need it. having it in the query with a custom hook for accessing makes it super easy for leaf components to access it.